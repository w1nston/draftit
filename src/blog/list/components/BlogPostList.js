import React, { Fragment } from 'react';
import format from 'date-fns/format';
import { Link } from 'react-router-dom';
import { css } from 'emotion';

const tableStyle = css`
  border-collapse: collapse;
`;

const tableHeaderCellStyle = css`
  padding: 12px 6px;
  text-align: left;
`;

const tableHeaderRowStyle = css`
  border-bottom: 1px solid #111;
`;

const tableRowStyle = css`
  &:nth-child(odd) {
    background-color: #dafbf2;
  }
`;

const tableCellStyle = css`
  padding: 16px 6px;
  text-align: left;
`;

const tableCellCenterStyle = css`
  padding: 16px 6px;
  text-align: center;
`;

const deleteButtonStyle = css`
  cursor: pointer;
`;

const publishButtonStyle = css`
  cursor: pointer;
`;

const BlogPostList = ({ posts = [] }) => (
  <table className={tableStyle}>
    <thead>
      <tr className={tableHeaderRowStyle}>
        <th className={tableHeaderCellStyle}>Title</th>
        <th className={tableHeaderCellStyle}>Created At</th>
        <th className={tableHeaderCellStyle}>Updated At</th>
        <th className={tableHeaderCellStyle}>Published At</th>
        <th className={tableHeaderCellStyle} />
      </tr>
    </thead>
    <tbody>
      {posts.map(post => (
        <tr className={tableRowStyle} key={post.title}>
          <td className={tableCellStyle}>{post.title}</td>
          <td className={tableCellCenterStyle}>
            {post.createdAt ? (
              format(post.createdAt, 'YYYY-MM-DD HH:mm')
            ) : (
              <Fragment>&mdash;</Fragment>
            )}
          </td>
          <td className={tableCellCenterStyle}>
            {post.updatedAt ? (
              format(post.updatedAt, 'YYYY-MM-DD HH:mm')
            ) : (
              <Fragment>&mdash;</Fragment>
            )}
          </td>
          <td className={tableCellCenterStyle}>
            {post.publishedAt ? (
              format(post.publishedAt, 'YYYY-MM-DD HH:mm')
            ) : (
              <Fragment>&mdash;</Fragment>
            )}
          </td>
          <td className={tableCellStyle}>
            Edit |{' '}
            <Link to={`/preview/${encodeURIComponent(post.title)}`}>Show</Link>{' '}
            | <button className={publishButtonStyle}>Publish</button> |{' '}
            <button
              className={deleteButtonStyle}
              onClick={() => props.onDelete && props.onDelete(post.title)}
            >
              Delete
            </button>
          </td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default BlogPostList;
