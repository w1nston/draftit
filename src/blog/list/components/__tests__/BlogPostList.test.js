import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import addDays from 'date-fns/add_days';
import BlogPostList from '../BlogPostList';

describe('BlogPostList', () => {
  it('renders with no blog posts', () => {
    const component = shallow(<BlogPostList />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with blog posts', () => {
    const posts = [{ title: 'Title', createdAt: undefined, publishedAt: undefined }];
    const component = shallow(<BlogPostList posts={posts} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
