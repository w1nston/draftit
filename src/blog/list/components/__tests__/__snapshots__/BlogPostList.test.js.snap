// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostList renders with blog posts 1`] = `
<table
  className="css-1i2v0m9"
>
  <thead>
    <tr
      className="css-19ma4in"
    >
      <th
        className="css-om3ifv"
      >
        Title
      </th>
      <th
        className="css-om3ifv"
      >
        Created At
      </th>
      <th
        className="css-om3ifv"
      >
        Updated At
      </th>
      <th
        className="css-om3ifv"
      >
        Published At
      </th>
      <th
        className="css-om3ifv"
      />
    </tr>
  </thead>
  <tbody>
    <tr
      className="css-108yoyb"
      key="Title"
    >
      <td
        className="css-1djw1qp"
      >
        Title
      </td>
      <td
        className="css-1qznyyb"
      >
        <React.Fragment>
          —
        </React.Fragment>
      </td>
      <td
        className="css-1qznyyb"
      >
        <React.Fragment>
          —
        </React.Fragment>
      </td>
      <td
        className="css-1qznyyb"
      >
        <React.Fragment>
          —
        </React.Fragment>
      </td>
      <td
        className="css-1djw1qp"
      >
        Edit |
         
        <Link
          replace={false}
          to="/preview/Title"
        >
          Show
        </Link>
         
        | 
        <button
          className="css-1h5x3dy"
        >
          Publish
        </button>
         |
         
        <button
          className="css-1h5x3dy"
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`BlogPostList renders with no blog posts 1`] = `
<table
  className="css-1i2v0m9"
>
  <thead>
    <tr
      className="css-19ma4in"
    >
      <th
        className="css-om3ifv"
      >
        Title
      </th>
      <th
        className="css-om3ifv"
      >
        Created At
      </th>
      <th
        className="css-om3ifv"
      >
        Updated At
      </th>
      <th
        className="css-om3ifv"
      >
        Published At
      </th>
      <th
        className="css-om3ifv"
      />
    </tr>
  </thead>
  <tbody />
</table>
`;
